components/ai_chat/resources/page/components/full_screen/index.tsx

```diff
@@ -0,0 +1,129 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Icon from '@brave/leo/react/icon'
+import styles from './style.module.scss'
+import Main from '../main'
+import { SidebarHeader } from '../header'
+import SidebarNav from '../sidebar_nav'
+import { useAIChat } from '../../state/ai_chat_context'
+import useMediaQuery from '$web-common/useMediaQuery'
+
+export default function FullScreen() {
+  const aiChatContext = useAIChat()
+  const asideAnimationRef = React.useRef<Animation | null>()
+  const controllerRef = React.useRef(new AbortController())
+  const isSmall = useMediaQuery('(max-width: 1024px)')
+  const [isOpen, setIsOpen] = React.useState(isSmall)
+  const [shouldRender, setShouldRender] = React.useState(!isSmall)
+
+  const initAsideAnimation = React.useCallback((node: HTMLElement | null) => {
+    if (!node) return
+    const open = { width: '340px', opacity: 1 }
+    const close = { width: '0px', opacity: 0 }
+    const animationOptions: KeyframeAnimationOptions = {
+      duration: 200,
+      easing: 'ease-out',
+      fill: 'forwards'
+    }
+    asideAnimationRef.current = new Animation(
+      new KeyframeEffect(node, [open, close], animationOptions)
+    )
+
+    // Make sure we're in the right state for our screen size when
+    asideAnimationRef.current.playbackRate = isSmall ? 1 : -1
+    asideAnimationRef.current.finish()
+  }, [])
```

Comment:
@petemill in a followup I'd like to revisit how the sidebar works, because its kinda painful to use

Reply:
> how ~the sidebar~ navigation works

👍